pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -352,7 +362,7 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
 			instance.Endpoint.ServicePortName = listenPort.Name
 			instance.Endpoint.EndpointPort = uint32(port)
 
-			localCluster := cb.buildInboundClusterForPortOrUDS(nil, int(ingressListener.Port.Number), endpointAddress, instance, nil)
+			localCluster := cb.buildInboundClusterForPortOrUDS(cb.proxy, int(ingressListener.Port.Number), endpointAddress, instance, nil)
```

Comment:
I noticed this is the right fix, @Stono  you set sidecar ingress

Reply:
Yes, there are 2 fixes - this part fixes the sidecar.ingress part and the other change fixes multiple services